simpler_eda/numerical_eda.py

```diff
@@ -80,5 +74,146 @@ def numerical_plot(
                  title = " Horsepower vs Acceleration",
                  font_size = 10)
     """
-
-    return numerical_plot
+    # Defensive programming: Check if user provides valid inputs
+
+    # Ensure title variable is specified by the user
+    assert title is not None, "Variable title needs to be specified."
+
+    error_one = "TypeError: Data must be entered as a pandas dataframe."
+    assert isinstance(data, pd.DataFrame), error_one
+
+    error_two = "Wrong type! X-axis variable must be entered as a String."
+    assert isinstance(xval, str), error_two
+
+    error_three = "Wrong type! Y-axis variable must be entered as a String."
+    assert isinstance(yval, str), error_three
+
+    error_four = "TypeError: x_transform must be of type boolean."
+    assert isinstance(x_transform, bool), error_four
+
+    error_five = "TypeError: y_transform must be of type boolean."
+    assert isinstance(y_transform, bool), error_five
+
+    error_six = "TypeError: plot_type must be either 'scatter' or 'line'."
```

Comment:
Instead of TypeError, we may consider "InputValueError" or just keep the "plot_type must ...".

Reply:
Updated to InputValueError